spec/ameba/rule/lint/duplicate_when_condition_spec.cr

```diff
@@ -0,0 +1,45 @@
+require "../../../spec_helper"
+
+module Ameba::Rule::Lint
+  describe DuplicateWhenCondition do
+    subject = DuplicateWhenCondition.new
+
+    it "passes if there are no duplicated `when` conditions" do
```

Comment:
nitpick: Could be nice to have a test with `in`

Reply:
That's not necessary since `in` branches can't be duplicates in exhaustive `case` variant.